### ISSUE TITLE
Mod helper getting an error and ingame mod settings are empty

### ISSUE THREAD

**Issue by Mani-cwaf** [gurrenm3/BTD-Mod-Helper#16]
I am not able to save my game (progress made after updating resets when game is closed, game saves also reset), and there are no mod settings available, I checked the log and this is the only error, but when I opened btd6 in the morning and mod helper was doing a cleanup 26 files were shown as corrupted/not recovered out of about a hundred

[10:40:00.151] [BloonsTD6_Mod_Helper] [ERROR] System.MissingMethodException: void Assets.Scripts.Unity.UI_New.Main.MainMenu.set_hasSeenModderWarning(bool)
  at MelonLoader.MelonHandler+<>c.<OnApplicationStart_Mods>b__30_0 (MelonLoader.MelonMod x) [0x0002a] in <67e16b0234254e2183a98652371f8451>:0
  at MelonLoader.MelonHandler.InvokeMelonMethod[T] (System.Collections.Generic.List`1[T]& melons, MelonLoader.MelonHandler+InvokeMelonMethodDelegate`1[T] method, System.Boolean remove_failed) [0x00032] in <67e16b0234254e2183a98652371f8451>:0
Setting breakpad minidump AppID = 960090
Steam_SetMinidumpSteamID:  Caching Steam ID:  76561199002699290 [API loaded no]

**Comment by doombubbles** [comment]
This is an issue that ought to have been fixed with v2.4.2. Can you confirm if this issue is still present for you using that version?

**Comment by YenixAlpha** [comment]
I can confirm this issue is still happening with the updated version

**Comment by doombubbles** [comment]
I'll need to see a complete log file then, as all parts of the Mod Helper referencing "MainMenu.set_hasSeenModderWarning" (something that was changed with v30.0) have been removed with v2.4.2. The files are in BloonTD6/MelonLoader/Logs for the 10 most recent logs, or just BloonTD6/MelonLoader/Latest.log for the most recent one.

**Comment by YenixAlpha** [comment]
[Latest.log](https://github.com/gurrenm3/BTD-Mod-Helper/files/8027171/Latest.log)

**Comment by MRS-OBAMA** [comment]
this seems to happen even when the mod helper dll is removed. seems like a BTD6 - melonloader issue?

**Comment by randomguyyyy** [comment]
i have this issue as well im going to reinstall melon loader

**Comment by Mani-cwaf** [comment]
I deleted the version.ddl, melonloader folder, and mods folder and everything is going fine, I still think there should be a fix to play mods on v30 though

**Comment by M4ng088** [comment]
But there is still a problem with saving your profil data, so when i complete an achievement, it sets me back to when i started playing today.

**Comment by randomguyyyy** [comment]
yeah i did see that after i went and it opened by  daily reward and got the new skin it reset everytime i left
